src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+module Security
+  # Class for handling SELinux kernel params
+  class SelinuxConfig
+    extend Yast::I18n
+    include Yast::I18n
+    include Yast::Logger
+
+    Yast.import "Bootloader"
+    Yast.import "ProductFeatures"
+
+    # @return [Security::SelinuxConfig::Mode] the last set mode, which can be differrent to the
+    #   {#running_mode} and {#configured_mode}. A call to {#save} is needed to make it the
+    #   {#configured_mode} for the next boot.
+    attr_reader :mode
+
+    # Constructor
+    def initialize
+      @mode = initial_mode
+    end
+
+    # Returns the initial SELinux mode
+    #
+    # It can be the proposed one if running in an installation or the configured mode for a
+    # running system
+    #
+    # @return [Security::SelinuxConfig::Mode]
+    def initial_mode
+      return @initial_mode if @initial_mode
+
+      propose! if Yast::Mode.installation
+
+      @initial_mode = configured_mode
+    end
+
+    # Returns the mode applied in the running system
+    #
+    # @note the system can be booted with a different SELinux mode that the configured one. To
+    #  know the running mode getenforce tool is used.
+    #
+    # @return [Mode] running SELinux mode if command executed successfully; :disabled otherwise
+    def running_mode
+      id = Yast::Execute.locally!(GETENFORCE_PATH, stdout: :capture).chomp.downcase.to_sym
+      Mode.find(id)
+    rescue Cheetah::ExecutionFailed => e
+      log.info(e.message)
+
+      Mode.find(:disabled)
+    end
+
+    # Returns a collection holding all known SELinux modes
+    #
+    # @return [Array<Security::SelinuxConfig::Mode>] a collection of knwon SELinux modes
+    def modes
+      Mode.all
+    end
+
+    # Set the mode to given value
+    #
+    # @note using nil means to set SELinux mode as disabled.
+    #
+    # @param id [String, Symbol, nil] a SELinux mode identifier
+    # @return [Mode, nil] found SelinuxConfig::Mode by given id; nil if none
```

Comment:
assignment does not have return.

lets try it youself:
```
jreidinger@50-3E-AA-D7-45-9F:~/prace/yast/security/src> irb
irb(main):001:0> def a=(test);false;end
=> :a=
irb(main):002:0> self.a = 15
=> 15
```

Reply:
it always return parameter

Reply:
Arrrrg!!! Stupid mistake